genesyscloud/conversations_messaging_integrations_instagram/genesyscloud_conversations_messaging_integrations_instagram_proxy.go

```diff
@@ -0,0 +1,168 @@
+	return p.deleteConversationsMessagingIntegrationsInstagramAttr(ctx, p, id)
+}
+
+// createConversationsMessagingIntegrationsInstagramFn is an implementation function for creating a Genesys Cloud conversations messaging integrations instagram
+func createConversationsMessagingIntegrationsInstagramFn(ctx context.Context, p *conversationsMessagingIntegrationsInstagramProxy, conversationsMessagingIntegrationsInstagram *platformclientv2.Instagramintegrationrequest) (*platformclientv2.Instagramintegration, *platformclientv2.APIResponse, error) {
+	return p.conversationsApi.PostConversationsMessagingIntegrationsInstagram(*conversationsMessagingIntegrationsInstagram)
+}
+
+// getAllConversationsMessagingIntegrationsInstagramFn is the implementation for retrieving all conversations messaging integrations instagram in Genesys Cloud
+func getAllConversationsMessagingIntegrationsInstagramFn(ctx context.Context, p *conversationsMessagingIntegrationsInstagramProxy) (*[]platformclientv2.Instagramintegration, *platformclientv2.APIResponse, error) {
+	var allInstagramIntegrationRequests []platformclientv2.Instagramintegration
+	const pageSize = 100
+
+	instagramIntegrationRequests, resp, err := p.conversationsApi.GetConversationsMessagingIntegrationsInstagram(pageSize, 1, "", "", "")
+	if err != nil {
+		return nil, resp, fmt.Errorf("Failed to get instagram integration request: %v", err)
+	}
+	if instagramIntegrationRequests.Entities == nil || len(*instagramIntegrationRequests.Entities) == 0 {
+		return &allInstagramIntegrationRequests, resp, nil
+	}
+	for _, instagramIntegrationRequest := range *instagramIntegrationRequests.Entities {
+		allInstagramIntegrationRequests = append(allInstagramIntegrationRequests, instagramIntegrationRequest)
+	}
+
+	for pageNum := 2; pageNum <= *instagramIntegrationRequests.PageCount; pageNum++ {
+		instagramIntegrationRequests, resp, err := p.conversationsApi.GetConversationsMessagingIntegrationsInstagram(pageSize, pageNum, "", "", "")
+		if err != nil {
+			return nil, resp, fmt.Errorf("Failed to get instagram integration request: %v", err)
+		}
+
+		if instagramIntegrationRequests.Entities == nil || len(*instagramIntegrationRequests.Entities) == 0 {
+			break
+		}
+
+		for _, instagramIntegrationRequest := range *instagramIntegrationRequests.Entities {
+			allInstagramIntegrationRequests = append(allInstagramIntegrationRequests, instagramIntegrationRequest)
+		}
+	}
+
+	return &allInstagramIntegrationRequests, resp, nil
+}
+
+// getConversationsMessagingIntegrationsInstagramIdByNameFn is an implementation of the function to get a Genesys Cloud conversations messaging integrations instagram by name
+func getConversationsMessagingIntegrationsInstagramIdByNameFn(ctx context.Context, p *conversationsMessagingIntegrationsInstagramProxy, name string) (id string, retryable bool, response *platformclientv2.APIResponse, err error) {
+	instagramIntegrationRequests, resp, err := getAllConversationsMessagingIntegrationsInstagramFn(ctx, p)
```

Comment:
I would use `p.getAllConversationsMessagingIntegrationsInstagram(ctx)` instead of the `Fn` function. It makes no difference now, but if we want to implement unit testing later it will make it easier to mock out this call.